casper/src/main/resources/PoS.rhox

```diff
@@ -191,21 +192,30 @@ new  PoS,
             } |
 
             new currentDeployerData in {
-              contract PoS (@"chargeDeploy", @deployerId, @amount, return) = {
-                new depositCh, tmpCh in {
-                  getMVar!(*stateCh, *tmpCh) |
-                  for (@state <- tmpCh) {
-                    deposit!(deployerId, amount, state.get("activeValidators"), *depositCh) |
-                    for(@depositResult <- depositCh){
-                      match depositResult {
-                        (true, _) => {
-                          currentDeployerData!((deployerId, amount)) |
-                          return!(depositResult)
-                        }
-                        (_, errorMessage) => {
-                          return!((false, "Deploy payment failed: " ++ errorMessage))
+              contract PoS (@"chargeDeploy", @deployerId, @amount, @sysAuthToken, return) = {
+                new isValidTokenCh in {
+                  sysAuthTokenOps!("check", sysAuthToken, *isValidTokenCh) |
+                  for (@isValid <- isValidTokenCh) {
+                    if (isValid) {
+                      new depositCh, tmpCh in {
+                        getMVar!(*stateCh, *tmpCh) |
+                        for (@state <- tmpCh) {
+                          deposit!(deployerId, amount, state.get("activeValidators"), *depositCh) |
+                          for(@depositResult <- depositCh){
+                            match depositResult {
+                              (true, _) => {
+                                currentDeployerData!((deployerId, amount)) |
+                                return!(depositResult)
+                              }
+                              (_, errorMessage) => {
+                                return!((false, "Deploy payment failed: " ++ errorMessage))
+                              }
+                            }
+                          }
                         }
                       }
+                    } else {
+                      return!(Nil)
```

Comment:
Shouldn't this return the same Either type e.g. `return!((false, "Invalid auth token"))`? In other methods also.

Reply:
Done.